core/fetcher.go

```diff
@@ -33,6 +36,16 @@ func NewBlockFetcher(client Client) *BlockFetcher {
 	}
 }
 
+func (f *BlockFetcher) Stop(ctx context.Context) error {
+	f.cancel()
+	select {
+	case <-f.doneCh:
+		return nil
+	case <-ctx.Done():
+		return fmt.Errorf("fetcher: unsubscribe from new block events: %w", ctx.Err())
```

Comment:
[question for reviewers]
not sure if this is the right way to stop the resources in here. I guess we can delete this part. What do you think?

Reply:
what about closing the underlying connection? What's responsible for that?

Reply:
good catch. I can support closing the connection if we make the `Client` a struct. would that be something we want?

Reply:
a033feb98bbe57abad444a34d63e8f72369fd0cd